Tests/SentryTests/SentryClientTests.swift

```diff
@@ -1522,29 +1531,28 @@ class SentryClientTest: XCTestCase {
         }
     }
     
-    private func assertValidErrorEvent(_ event: Event, _ error: NSError, exceptionValue: String? = nil) {
+    private func assertValidErrorEvent(_ event: Event, _ expectedError: NSError, exceptionValue: String? = nil) throws {
         XCTAssertEqual(SentryLevel.error, event.level)
-        XCTAssertEqual(error, event.error as NSError?)
+        XCTAssertEqual(expectedError, event.error as NSError?)
         
         guard let exceptions = event.exceptions else {
             XCTFail("Event should contain one exception"); return
         }
         XCTAssertEqual(1, exceptions.count)
         let exception = exceptions[0]
-        XCTAssertEqual(error.domain, exception.type)
+        XCTAssertEqual(expectedError.domain, exception.type)
         
-        XCTAssertEqual(exceptionValue ?? "Code: \(error.code)", exception.value)
+        XCTAssertEqual(exceptionValue ?? "Code: \(expectedError.code)", exception.value)
         
         XCTAssertNil(exception.threadId)
         XCTAssertNil(exception.stacktrace)
-        
-        guard let mechanism = exception.mechanism else {
-            XCTFail("Exception doesn't contain a mechanism"); return
-        }
+
+        let mechanism = try XCTUnwrap(exception.mechanism)
+        let meta = try XCTUnwrap(mechanism.meta)
+        let actualError = try XCTUnwrap(meta.error)
         XCTAssertEqual("NSError", mechanism.type)
-        XCTAssertNotNil(mechanism.meta?.error)
```

Comment:
This gives us a more granular idea of which thing is missing; It is responsible for all of the `throws` additions in this file.